genomic_address_service/classes/assign.py

```diff
@@ -153,31 +184,29 @@ def get_threshold_idx(self,dist):
         return 0
 
     def guess_file_type(self,f):
-        file_type = None
-        for t in EXTENSIONS:
-            for e in EXTENSIONS[t]:
-                if e in f:
-                    file_type = t
-                    break
-        return file_type
+        extension = os.path.splitext(f)[1]
+        valid_extensions = list(EXTENSIONS.keys())
+
+        if not extension in valid_extensions:
+            message = f'{f} does not have a valid extension {valid_extensions}'
+            raise Exception(message)
+
+        return EXTENSIONS[extension]
 
-    def read_data(self,f):
+    def read_data(self, f):
         df = pd.DataFrame()
         file_type = self.guess_file_type(f)
-        if file_type == 'text':
+
+        if file_type == TEXT:
```

Comment:
You should also test if the TEXT file has delimiter other than `\t` tab delimiter. In that case pandas will not be able to identify columns so we can then fail the input saying that expected delimiter was not found. Are we supporting also csv files?

Reply:
See here: https://github.com/phac-nml/genomic_address_service/pull/38#issuecomment-2790493569